### PR TITLE
Allow oc extract to output to stdout

### PR DESCRIPTION
**Pull request by smarterclayton**
oc extract configmap/foo --to=-

will output the configmap to stdout.

### CODE REVIEW

**Comment by smarterclayton** [comment]
@openshift/sig-user-interface

File: pkg/oc/cli/cmd/extract.go [hunk 1]
```diff
@@ -38,6 +41,9 @@ var (
 	  # extract the config map "nginx" to the /tmp directory
 	  %[1]s extract configmap/nginx --to=/tmp
 
+		# extract the config map "nginx" to STDOUT
+	  %[1]s extract configmap/nginx --to=-
```
Comment (mfojtik) [inline pkg/oc/cli/cmd/extract.go]: this seems weird, can't we just stream to stdout by default when `--to` is not set?

Comment (smarterclayton) [reply pkg/oc/cli/cmd/extract.go]: We could, would you expect that?  Extract with STDOUT is more of a viewing action (like `kubectl view`), while extract to file is more "normal".

Comment (mfojtik) [reply pkg/oc/cli/cmd/extract.go]: @smarterclayton how about `--to-stdout` ?

Comment (smarterclayton) [reply pkg/oc/cli/cmd/extract.go]: We use '-' pretty consistently.  Hrm... I guess I could do no arg.

Comment (smarterclayton) [reply pkg/oc/cli/cmd/extract.go]: Hrm, no-arg today means "extract to current directory".  So if we change the behavior of no arg it's not backwards compatible.

Given that we use '-' in other places it's a pretty reasonable convention to continue.  Do we have any examples of --to-stdout or similar?

Comment (juanvallejo) [reply pkg/oc/cli/cmd/extract.go]: Per our conversation, and given that conventions do exist to have `-` mean "output to stdin/stdout", I think it is reasonable to have `-` mean stdout in this case.

**Comment by smarterclayton** [comment]
The challenge with extra flags is that you have to then deal with mutual incompatibility, which is more code and validation.  By combining them, you remove the possibility for that.

Any other opinions?

**Comment by smarterclayton** [comment]
I'm open to follow up suggestions, hearing no strong dissent.......

**Pull request merged**